### PR TITLE
[shopsys] added warnings about version to installation guides

### LINKED ISSUE #794: Provide more information about current version in installation guide
### What is happening
From time to time, we perform BC breaking changes in our installation guides, even though we have a process that should ensure avoiding this.
See e.g. https://github.com/shopsys/shopsys/pull/737

And now once again, in the installation guide, I see an instruction to run
```
./scripts/install.sh
```
however, this is not released yet so I am not able to do that when installing `project-base` from the last beta version (`beta5`) :disappointed: 

### Expected result
As our process with writing TODO comments in the documentation does not work properly, I suggest adding a simple note to the beginning of each installation guide, that would say: "Beware, you are viewing installation guide for the current master. Installation guide for the last release might be more relevant for you." where "Installation guide for the last release" would be a link to the last released version of the guide. This could be modified automatically during the release process. I do not insist on this particular words, however, the idea is clear IMHO.

### PR DESCRIPTION
**Pull request by vitek-rostislav**
| Q             | A
| ------------- | ---
|Description, reason for the PR| On Github, the installation guide version is not clear at a first sight
|New feature| No 
|[BC breaks](/docs/contributing/backward-compatibility-promise.md)| No 
|Fixes issues| closes #794 
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: utils/releaser/src/ReleaseWorker/ReleaseCandidate/SetReleasedVersionNoteInInstallationGuidesReleaseWorker.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+     */
+    public function work(Version $version): void
+    {
+        $this->symfonyStyle->note('Setting note about the released version in installation guides.');
+        $fileInfos = $this->installationGuideFilesProvider->provide();
+        foreach ($fileInfos as $fileInfo) {
+            $newContent = $this->installationGuideFileManipulator->setReleasedVersionNote($fileInfo, $version);
+            FileSystem::write($fileInfo->getPathname(), $newContent);
+        }
+        $this->commit('docker installation guides: updated note about the version');
```
Comment (boris-brtan) [inline utils/releaser/src/ReleaseWorker/ReleaseCandidate/SetReleasedVersionNoteInInstallationGuidesReleaseWorker.php]:
```suggestion
        $this->commit('docker installation guides: updated note about the released version');
```

File: utils/releaser/src/ReleaseWorker/AfterRelease/SetUnreleasedNoteInInstallationGuidesReleaseWorker.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+     */
+    public function work(Version $version): void
+    {
+        $this->symfonyStyle->note('Setting note about the unreleased version in installation guides.');
+        $fileInfos = $this->installationGuideFilesProvider->provide();
+        foreach ($fileInfos as $fileInfo) {
+            $newContent = $this->installationGuideFileManipulator->setUnreleasedNote($fileInfo, $version);
+            FileSystem::write($fileInfo->getPathname(), $newContent);
+        }
+        $this->commit('docker installation guides: updated note about the version');
```
Comment (boris-brtan) [inline utils/releaser/src/ReleaseWorker/AfterRelease/SetUnreleasedNoteInInstallationGuidesReleaseWorker.php]:
```suggestion
        $this->commit('docker installation guides: updated note about the unreleased version');
```

File: utils/releaser/src/FileManipulator/InstallationGuideFileManipulator.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+use Nette\Utils\Strings;
+use PharIo\Version\Version;
+use Symfony\Component\Finder\SplFileInfo;
+
+class InstallationGuideFileManipulator
+{
+    /**
+     * @var string
+     */
+    private const UNRELEASED_NOTE_REGEX = '#^\*\*This guide is for version which is not released yet\. See the \[version for `(.*)`\]\(https:\/\/github\.com\/shopsys\/shopsys\/blob\/(.*)\/docs\/installation\/.*\.md\)\.\*\*$#m';
```
Comment (boris-brtan) [inline utils/releaser/src/FileManipulator/InstallationGuideFileManipulator.php]:
```suggestion
    private const UNRELEASED_NOTE_REGEX = '#^\*{2}This guide is for version which is not released yet\. See the \[version for `(.*)`\]\(https:\/\/github\.com\/shopsys\/shopsys\/blob\/(.*)\/docs\/installation\/.*\.md\)\.\*{2}$#m';
```

File: docs/installation/installation-using-docker-linux.md [hunk 4]
```diff
@@ -1,5 +1,7 @@
 # Installation Using Docker for Linux
 
+**This guide is for version which is not released yet. See the [version for `v7.0.0-beta5`](https://github.com/shopsys/shopsys/blob/v7.0.0-beta5/docs/installation/installation-using-docker-linux.md).**
```
Comment (boris-brtan) [inline docs/installation/installation-using-docker-linux.md]: `for version` or `for the version` ?

File: utils/releaser/src/FilesProvider/InstallationGuideFilesProvider.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+final class InstallationGuideFilesProvider
+{
+    /**
+     * @return \Symfony\Component\Finder\SplFileInfo[]
+     */
+    public function provide(): array
+    {
+        $finder = Finder::create()->files()
+            ->name('installation-using*.md')
```
Comment (boris-brtan) [inline utils/releaser/src/FilesProvider/InstallationGuideFilesProvider.php]: so what about production installation document ???

File: docs/installation/installation-using-docker-linux.md (hunk 4 above)
Comment (vitek-rostislav) [reply docs/installation/installation-using-docker-linux.md]: Right, missing article :slightly_smiling_face:

File: utils/releaser/src/FilesProvider/InstallationGuideFilesProvider.php (hunk 5 above)
Comment (vitek-rostislav) [reply utils/releaser/src/FilesProvider/InstallationGuideFilesProvider.php]: it was not in a scope of this PR :innocent:

**Comment by vitek-rostislav** [comment]
Hi @boris-brtan, thanks for review and suggestions. Should be ok now

**Comment by boris-brtan** [comment]
### TESTS
i just add one confusion into the pool of confusions. Is that version v7.0.0-betaX or it is version 7.0.0-betaX ???

**Comment by vitek-rostislav** [comment]
Hi, thanks! 
> Is that version v7.0.0-betaX or it is version 7.0.0-betaX ???

Well, tag always includes the `v` prefix, so you use it this way in the release command argument as well. However, in composer.json, the `v` prefix should be omitted there
https://github.com/shopsys/shopsys/blob/master/docs/contributing/releasing-a-new-version-of-shopsys-framework.md#notes

File: utils/releaser/src/FilesProvider/InstallationGuideFilesProvider.php (hunk 5 above)
Comment (boris-brtan) [reply utils/releaser/src/FilesProvider/InstallationGuideFilesProvider.php]: sorry but scope or not scope......THIS IS CRUCIAL

**Comment by vitek-rostislav** [comment]
> production docs please

ok :slightly_smiling_face:

**Pull request merged**